rust-runtime/aws-smithy-async/src/future/pagination_stream.rs

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Types to support stream-like operations for paginators.
+
+use crate::future::rendezvous;
+use pin_project_lite::pin_project;
+use std::fmt;
+use std::future::poll_fn;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pub mod collect;
```

Comment:
a meta nit, can you put `fn_stream` back in it's own module? that would make this much easier to review

Reply:
Yep, updated in 1add37f.